src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -43,12 +42,14 @@
 import org.opensearch.security.user.AuthCredentials;
 import org.opensearch.security.util.KeyUtils;
 
-import static org.opensearch.security.OpenSearchSecurityPlugin.LEGACY_OPENDISTRO_PREFIX;
-import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
-import static org.opensearch.security.util.AuthTokenUtils.isAccessToRestrictedEndpoints;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtParser;
+import io.jsonwebtoken.JwtParserBuilder;
+import io.jsonwebtoken.security.WeakKeyException;
 
 public class OnBehalfOfAuthenticator implements HTTPAuthenticator {
 
+    private static final int MINIMUM_SIGNING_KEY_BIT_LENGTH = 512;
```

Comment:
Do we have a reference from this number? It seems fine but don't know where it came from

Reply:
Good question, I had a little bit of a hard time articulating where this value came from other than I copied it from another error message that said the value was associated with the HS512 algorithm.

_ChatGPT helped me answer this with:_
> "The 512 value aligns with the HS512 algorithm's requirement for a secure hash-based message authentication code"